pkg/agent/openflow/pipeline.go

```diff
@@ -3011,15 +3007,15 @@ func (f *featureService) gatewaySNATFlows() []binding.Flow {
 			pktDstRegMarks = append(pktDstRegMarks, ToUplinkRegMark)
 		}
 		for _, pktDstRegMark := range pktDstRegMarks {
-			// This generates the flow to match the first packet of NodePort / LoadBalancer connection initiated through the
-			// Antrea gateway and externalTrafficPolicy of the Service is Cluster, and the selected Endpoint is on a remote
-			// Node, then ConnSNATCTMark will be loaded in DNAT CT zone, indicating that SNAT is required for the connection.
+			// This generates the flow to match the first packets of externally-originated connections towards external
+			// addresses of the Service initiated through the Antrea gateway, and the selected Endpoint is on a remote Node,
+			// then ConnSNATCTMark will be loaded in DNAT CT zone, indicating that SNAT is required for the connection.
 			flows = append(flows, SNATMarkTable.ofTable.BuildFlow(priorityNormal).
 				Cookie(cookieID).
 				MatchProtocol(ipProtocol).
 				MatchCTStateNew(true).
 				MatchCTStateTrk(true).
-				MatchRegMark(FromGatewayRegMark, pktDstRegMark, ToClusterServiceRegMark).
+				MatchRegMark(FromGatewayRegMark, pktDstRegMark, ExternallyAccessibleRegMark).
```

Comment:
It woudl be more readable if this is `MatchRegMark(FromGatewayRegMark, pktDstRegMark, ToExternalAddressRegMark).`

Reply:
You are right, updated as you suggested.